src/app/courses/courses-list/courses-list.component.ts

```diff
@@ -1,67 +1,27 @@
 import { FilterByTitlePipe } from '../courses-list-item/filter-by-title.pipe';
+import { CoursesService } from '../../services/courses.service';
 
 @Component({
   selector: 'vp-courses-list',
   templateUrl: './courses-list.component.html',
   styleUrls: ['./courses-list.component.css']
 })
 export class CoursesListComponent implements OnInit {
-  public byTitleFilter: string;
   public coursesItems: CoursesListItem[];
 
-  constructor(private filter: FilterByTitlePipe) { }
+  constructor(private coursesService: CoursesService) { }
 
   ngOnInit() {
-    this.coursesItems = this.getCoursesList();
+    this.coursesItems = this.coursesService.getList();
   }
 
   deleteCourseById(event: number) {
-    console.log('Deleting course by #id:' + event);
-  }
-
-  getCoursesList(): CoursesListItem[] {
-    return [
-      {
-        id: 1,
-        title: 'Video course #1 Intro',
-        creationDate: new Date ('2020-04-11'),
-        duration: 88,
-        description: 'Description of the Video course #1 Learn about where you can find course descriptions, what information they include'
-          + ', how they work, and details about various components of a course description. Course descriptions report information about a'
-          + ' university or college\'s classes. They\'re published both in course catalogs that outline degree requirements and in course'
-          + ' schedules that contain descriptions for all courses offered during a particular semester',
-        topRated: true
-      },
-      {
-        id: 2,
-        title: 'Video course #2 Components',
-        creationDate: new Date ('2019-12-02'),
-        duration: 48,
-        description: 'Description of the Video course #2 Learn about where you can find course descriptions, what information they include'
-          + ', how they work, and details about various components of a course description.',
-        topRated: false
-      },
-      {
-        id: 3,
-        title: 'Video course #3 Unit Testing',
-        creationDate: new Date ('2018-11-09'),
-        duration: 75,
-        description: 'Description of the Video course #3  Course descriptions report information about a university or college\'s classes.'
-          + ' They\'re published both in course catalogs that outline degree requirements and in course schedules that contain'
-          + ' descriptions for all courses offered during a particular semester',
-        topRated: true
-      },
-      {
-        id: 4,
-        title: 'Video course #4 Directives and Pipes',
-        creationDate: new Date ('2019-05-09'),
-        duration: 63,
-        description: 'Description of the Video course #4 They\'re published both in course catalogs that outline degree requirements'
-          + ' and in course schedules that contain descriptions for all courses offered during a particular semester',
-        topRated: false
-      }
-    ];
+    const confirm = window.confirm('Do you really want to delete this course?');
```

Comment:
it's bad practice using global variables such as window in the components, downside is it's difficult to test, you cannot really mock them
you think it will be there, but if for example you are using SSR there will be no window
better to make and injection token
```
const WINDOW = new InjectionToken('window', {
  provideIn: 'root',
  factory: function() {
    return window;
  },
});
```
and inject it into your component
`constructor(@Inject(WINDOW) private window: Window)`
in this case people who are trying to render your app with SSR can redefine token WINDOW and use mock instead of real window and you can test it properly by providing a mock instead of overriding real method on window